### ISSUE TITLE
Converting Keras model to CMSIS-NN

### ISSUE THREAD

**Issue by phanichinimilli** [ARM-software/CMSIS_5#1229]
I'm trying to implement a CMSIS-NN model for sign language classification of numbers(0-9), I have started by following the [sign language example](https://github.com/uva-linklab/CMSIS_NN_nrf52840), but the implementer aforementioned that inference is not working.
All I'm getting web sources to convert a caffe model to CMSIS-NN, but not much help with Keras mode to CMSIS-NN.
So I've used MNIST [example](https://github.com/autoas/CMSIS_5/blob/develop/CMSIS/NN/Examples/ARM/arm_nn_examples/mnist/mnist-on-keras.ipynb) and modified it to work for sign language which is as below [https://github.com/phanichinimilli/cmsis_nn_study/blob/main/py_scripts/sign_lang_keras.ipynb](https://github.com/phanichinimilli/cmsis_nn_study/blob/main/py_scripts/sign_lang_keras.ipynb).

I can see good accuracy for Keras model , but the quantized CMSIS-NN seems to be have bad accuracy i.e. <50%.
I've referred this [guide](https://github.com/phanichinimilli/cmsis_nn_study/blob/main/py_scripts/sign_lang_keras.ipynb) , it sure have made my understanding better, but to the extent of improving accuracy. 
I do have following unknowns 
1. Preprocessing of input image is dividing by 255.0, I'm using preprocessed test data , which is in Qmn format and feeding it to CMSIS-NN code. My assumption is that it should not make any difference, but I see some of the CIFAR10 implementations do use pre-processing of input data in CMSIS-NN code.
2. Does the shift calculations change if preprocessing of input is done in CMSIS-NN code, I mean I dont have to consider fi for first convolution layer ?
The bias shift is : (fi + fw) - fb
The out shift is : (fi + fw) - fo
3. I'm not sure where I went wrong , any kind of help would be appreciated.

**Comment by felix-johnny** [comment]
@phanichinimilli Thanks for the question. The example that you point out uses the legacy API's(https://github.com/ARM-software/CMSIS_5/tree/develop/CMSIS/NN#legacy-vs-tfl-micro-compliant-apis) for which support is minimal. 

  The focus in CMSIS-NN is on int8 kernels as per TFLM's quantization spec(https://www.tensorflow.org/lite/performance/quantization_spec). Could you maybe try something like what is mentioned in https://www.tensorflow.org/lite/convert/#convert_a_keras_model_ . That should give you a way to convert a Keras model into a TFLM model which can then utilize the optimizations in CMSIS-NN.  this is mainly for point 2 though.
